### PR TITLE
[Slice-631381] Add Mark as Accepted in posted documents for Spanish SII

### PR DESCRIPTION
**Pull request by Explorer986**
## What & why

Documents can get stuck in an Incorrect or Pending SII status and there was no way to clear that from the posted document itself. Only the SII History page could mark a document as accepted.

This adds a Mark As Accepted action to the posted sales, purchase and service invoice and credit memo pages. It does the same thing the SII History page already does, so users can correct a stuck document without leaving the page they are on.

The action resolves the SII state by document source, type and number to match the SII Status flowfield. Earlier attempts filtered on the SII posting date too, which meant documents whose stored SII posting date differed from the document
date (for example corrected credit memos) would not resolve and the click did nothing. The page owns the message shown to the user and the codeunit just returns whether it succeeded.

The action is gated behind the SII enabled check and the Show Advanced Actions setting, same as the other advanced SII actions.

## Linked work

Fixes [AB#631381](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/631381)

## How I validated this

- [x] I read the full diff and it contains only changes I intended.
- [x] I built the affected app(s) locally with no new analyzer warnings.
- [x] I ran the change in Business Central and confirmed it behaves as expected.
- [x] I added or updated tests for the new behavior, or explained below why none are needed.

**What I tested and the outcome**

- Built Base Application and Tests-Local for ES locally.
- Marked a posted document that was in Incorrect status from the posted page and confirmed it moved to Accepted With Errors, matching the SII History behavior.
- Confirmed a document that was already accepted shows the "already marked as accepted" message and its status is unchanged.
- Confirmed a document with no SII state shows the no status message.
- Added 9 tests in SIIDocumentProcessing.Codeunit.al: happy path for each posted document type, the already accepted case, the no state case, and 6 UI tests that open the real posted pages and invoke the action.

## Risk & compatibility

Low. The change is additive: a new action on existing posted pages plus a new public method on the SII Management codeunit. No schema, upgrade or data changes.
The required table permissions (SII Doc. Upload State and SII History) are already granted in the SIILOCAL permission set. The action is only visible when SII is enabled and Show Advanced Actions is on.

### CODE REVIEW

File: src/Layers/ES/BaseApp/Local/SII/EServices/EDocument/SIIManagement.Codeunit.al [hunk 1]
```diff
@@ -940,6 +950,35 @@ codeunit 10756 "SII Management"
...
+    begin
+        SIIDocUploadState.SetRange("Document Source", DocumentSource);
+        SIIDocUploadState.SetRange("Document Type", DocumentType);
+        SIIDocUploadState.SetRange("Document No.", DocumentNo);
+        if not SIIDocUploadState.FindFirst() then begin
+            FeedbackMessage := NoSIIStateErr;
+            exit(false);
+        end;
+
+        SIIHistory.SetRange("Document State Id", SIIDocUploadState.Id);
```
[the remark this answers is not in the document]
Comment (Explorer986) [reply src/Layers/ES/BaseApp/Local/SII/EServices/EDocument/SIIManagement.Codeunit.al:966]: Manual one-click action on a single document, same Document State Id filter the rest of SII already uses. Adding a key would only slow the bulk-upload inserts, so leaving as is.

**Comment by alexei-dobriansky** [comment]
## Good Sense Reviewer - Round 1

# **Recommendation: Accept**

<details>
<summary><strong>What this PR does</strong></summary>

This adds a **Mark As Accepted** action to the posted sales, purchase, and service invoice and credit memo pages for Spanish SII. The action calls the existing SII marking logic through a new document-based entry point, so a user can fix an Incorrect or Pending SII status from the posted document page instead of opening SII History.

The document lookup matches the posted document SII Status flowfield by document source, document type, and document number. The action is shown only when advanced SII actions are enabled, and it is enabled only when SII is enabled.
</details>

<details>
<summary><strong>Problem-solution fit</strong></summary>

**Fit:** Strong

The change matches the requested slice. It reuses the existing SII History behavior, avoids filtering by posting date, and covers the posted document types that expose the SII status.
</details>

<details>
<summary><strong>Suggestions</strong></summary>

No suggestions.
</details>

<details>
<summary><strong>Risk assessment and necessity</strong></summary>

**Risk:** Low. The change is additive and limited to ES SII page actions plus one SII Management helper. The helper updates the same SII Doc. Upload State and SII History records as the existing advanced action, and the permission set already grants access to those tables.

**Necessity:** The action is needed so users can correct a stuck posted document status without navigating away to SII History. The added tests cover the document-based helper, the already-accepted and no-state paths, visibility and enabled setup flags, and all six posted page actions.
</details>

---
> [AI-PR-REVIEW] version=1 promptVersion=4 system=github pr=10735 round=1 by=alexei-dobriansky at=2026-08-27T19:37:00Z lastSha=c918fdcc1f9de525571fa8c8bfefb08d8b686284 reviewKey=d672ef9366fcad4a311c97468d1c7b674619ec722007f706540572e316fae1ab suggestions=none

**Comment by alexei-dobriansky** [comment]
## Good Sense Reviewer - Round 2

# **Recommendation: Accept**

<details>
<summary><strong>What this PR does</strong></summary>

This round keeps the posted-document **Mark As Accepted** flow and tightens the helper surface to internal use. It also adds coverage for the case where a matching SII document state exists but no SII history entry exists, so the helper returns the same missing-status message instead of trying to mark a non-existent history record.

The new guard fits the document lookup path: it checks for the state first, then checks for the latest history entry before calling the existing history marking logic. The updated test exercises that edge case directly, and the changed helper is still callable from the added posted page actions inside the same app.
</details>

<details>
<summary><strong>Status of previous suggestions</strong></summary>

No previous suggestions were open from round 1.
</details>

<details>
<summary><strong>New observations (commits since round 1)</strong></summary>

None.
</details>

<details>
<summary><strong>Risk assessment and necessity</strong></summary>

**Risk:** Low. The round-2 code change is limited to access scope on the helper and one extra no-history branch that returns a message instead of changing data. The original page actions still call the same SII Management path, and the added test covers the new branch.

**Necessity:** The added no-history guard is useful defensive handling for a document state that has no related history. Making the helper internal is also appropriate because this is page action plumbing, not a public integration surface.
</details>

---
> [AI-PR-REVIEW] version=1 promptVersion=4 system=github pr=10735 round=2 by=alexei-dobriansky at=2026-08-28T01:14:57Z lastSha=1a08369b74ba8dc91a9e42df3aae74330fd77396 reviewKey=3a35a0978257c7cf5f650ced8b8284e0f4ef18a783b48013695058af7b126bf7 suggestions=none parentRound=1

**Review by ventselartur — APPROVED** [review]

File: src/Layers/ES/Tests/Local/SIIDocumentProcessing.Codeunit.al [hunk 2]
```diff
@@ -1107,6 +1109,324 @@ codeunit 147522 "SII Document Processing"
           SIIHistory."Upload Type"::Regular, false);
     end;
 
+    [Test]
+    [Scope('OnPrem')]
+    procedure SIIMarkPostedDocumentAsAccepted()
+    var
+        SIIDocUploadState: Record "SII Doc. Upload State";
+    begin
+        // [FEATURE] [Advanced Mark]
```
Comment (ventselartur) [inline src/Layers/ES/Tests/Local/SIIDocumentProcessing.Codeunit.al:1118]: Please, try to use the AL test agent for your tests :)

Comment (Explorer986) [reply src/Layers/ES/Tests/Local/SIIDocumentProcessing.Codeunit.al:1118]: sure, from next time, I will use bc-agentic for generating tests for my changes

**Review by alexei-dobriansky — APPROVED** [review]
